src/server/scripts/EasternKingdoms/ScarletEnclave/chapter2.cpp

```diff
@@ -623,7 +623,8 @@ class npc_a_special_surprise : public CreatureScript
             ExecuteSpeech_Counter = 0;
             PlayerGUID.Clear();
 
-            me->SetImmuneToPC(false);
+            me->RemoveUnitFlag(UNIT_FLAG_PACIFIED);
+            me->SetUnitFlag(UNIT_FLAG_IMMUNE_TO_PC);
```

Comment:
```suggestion
            me->SetImmuneToPC(true);
```
This should be all that's required. This pacified unit flag doesn't get touched in sniffs either. We also prefer to use helper methods instead of setting/removing the raw flags ourselves.

Reply:
On second glance, maybe instead of using the pacified flag, just setting their react state to passive would accomplish a similar effect.